SPEC.md

```diff
@@ -98,61 +98,113 @@ A build tool is something which reads one or more input artifacts and writes one
 ## Specifications
 ### Artifact ID
 
-GitBOM uses the git object id (as a blob) for an artifact as its artifact ID.
+A hash function over the []byte of an artifact satisfy the canonical, unique, immutable requirements for an artifact identifier.
+
+The vast majority of the source code artifacts are already indexed by git object id (gitoids) as git objects of type blob.
+
+For this reason GitBOM uses the git object id (gitoid) (as a blob) for an artifact as its artifact ID.  
+
```

Comment:
It probably makes sense to have this as it's own subsection rather than hanging off of the "Artifact ID" subsection.
```suggestion
### Gitoid Identifier Types
```suggestion